mne/annotations.py

```diff
@@ -331,3 +331,140 @@ def _read_annotations(fid, tree):
+                     enumerate(annotations.description)
+                     if kk in event_id_]
+        if len(event_sel) == 0:
+            raise ValueError('Could not find any of the events you specified.')
+        values = [event_id_[kk] for kk in
+                  annotations.description[event_sel]]
+        previous_value = np.zeros(len(event_sel))
+        inds = inds[event_sel]
+        events = np.c_[inds, previous_value, values].astype(int)
+        return events, event_id_
+
+    elif convention == "eeglab":
+        types = annotations.description
+        latencies = annotations.onset
+
+        if "boundary" in types and "boundary" not in event_id:
+            warn("The data contains 'boundary' events, indicating data "
+                 "discontinuities. Be cautious of filtering and epoching "
+                 "around these events.")
+
+        if len(types) < 1:  # if there are 0 events, we can exit here
+            logger.info('No events found, returning empty stim channel ...')
+            return np.zeros((0, 3)), dict()
+
+        not_in_event_id = set(x for x in types if x not in event_id)
+        not_purely_numeric = set(x for x in not_in_event_id if not x.isdigit())
+        no_numbers = set([x for x in not_purely_numeric
+                          if not any([d.isdigit() for d in x])])
+        have_integers = set([x for x in not_purely_numeric
+                             if x not in no_numbers])
+        if len(not_purely_numeric) > 0:
+            basewarn = "Events like the following will be dropped"
+            n_no_numbers, n_have_integers = len(no_numbers), len(have_integers)
+            if n_no_numbers > 0:
+                no_num_warm = " entirely: {0}, {1} in total"
+                warn(basewarn + no_num_warm.format(list(no_numbers)[:5],
+                                                   n_no_numbers))
+            if n_have_integers > 0 and event_id_func is None:
+                intwarn = (", but could be reduced to their integer part "
+                           "instead with the default `event_id_func`: "
+                           "{0}, {1} in total")
+                warn(basewarn + intwarn.format(list(have_integers)[:5],
+                                               n_have_integers))
+
+        events = list()
+        for tt, latency in zip(types, latencies):
+            try:  # look up the event in event_id and if not, try event_id_func
+                ev_code = event_id[tt] if tt in event_id else event_id_func(tt)
+                events.append([int(latency), 1, ev_code])
+            except (ValueError, TypeError):  # if event_id_func fails
+                pass  # We're already raising warnings above, so we just drop
+
+        if len(events) < len(types):
+            missings = len(types) - len(events)
+            msg = ("{0}/{1} event codes could not be mapped to integers. Use "
+                   "the 'event_id' parameter to map such events manually.")
```

Comment:
this msg is about using event_id, but this branch can be visited if event_id was used but not all events matched. Maybe the check should be `if len(events) < len(types) and event_id is None`?